### PR TITLE
Rdm 4537 doc demo

### PR DESCRIPTION
**Pull request by adnanakgun**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-4537

### Change description ###

Added demos for workbasket and case list filters

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: demo/src/app/workbasket-filters-consumer.component.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (andrewfolga) [inline demo/src/app/workbasket-filters-consumer.component.ts]: The only extracted component we expose should be the new case list filters. I think we should remove workbasket filters from the demo app as discussed before.

File: src/shared/components/case-list-filters/case-list-filters.component.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<ng-container *ngIf="jurisdictions">
```
Comment (andrewfolga) [inline src/shared/components/case-list-filters/case-list-filters.component.html]: Case list filters should be reusing workbasket filters not copying it similar to search filters and search filters wrapper.

File: demo/src/app/workbasket-filters-consumer.component.ts (hunk 1 above)
Comment (adnanakgun) [reply demo/src/app/workbasket-filters-consumer.component.ts]: Done

File: src/shared/components/case-list-filters/case-list-filters.component.html (hunk 2 above)
Comment (adnanakgun) [reply src/shared/components/case-list-filters/case-list-filters.component.html]: Done

**Comment by adnanakgun** [comment]
Please re-review @andrewfolga

File: src/shared/components/case-list-filters/case-list-filters.component.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { READ_ACCESS } from '../../domain/case-view/access-types.model';
+import { DefinitionsService } from '../../services';
+import { Jurisdiction } from '../../domain';
+
+@Component({
+  selector: 'ccd-case-list-filters',
+  templateUrl: './case-list-filters.component.html'
+})
+export class CaseListFiltersComponent implements OnInit {
```
Comment (andrewfolga) [inline src/shared/components/case-list-filters/case-list-filters.component.ts]: Unit tests checking emitting apply and reset notifications?

Comment (adnanakgun) [reply src/shared/components/case-list-filters/case-list-filters.component.ts]: added tests

**Comment by adnanakgun** [comment]
Please re-review @andrewfolga

**Pull request merged**